app/controllers/corrections/CorrectPreviousReturnController.scala

```diff
@@ -56,10 +60,19 @@ class CorrectPreviousReturnController @Inject()(
           Future.successful(BadRequest(view(formWithErrors, mode, period))),
 
         value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
-            _              <- cc.sessionRepository.set(updatedAnswers)
-          } yield Redirect(CorrectPreviousReturnPage.navigate(mode, updatedAnswers))
-      )
+      for {
+        updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
+        periods <- returnStatusConnector.listStatuses(request.registration.commencementDate)
+        _ <- cc.sessionRepository.set(updatedAnswers)
+      } yield {
+        periods match {
+          case Right(periods) =>Redirect(CorrectPreviousReturnPage.navigate(mode, updatedAnswers, periods.size))
+          case Left(value) =>
+            logger.error(s"there was an error $value")
+            throw new Exception(value.toString)
```

Comment:
I think redirecting to journeyRecovery might be better here

Reply:
done